docs/adr/0015-use-css-variables-and-scss.md

```diff
@@ -0,0 +1,185 @@
+
+#### SASS variables
+
+[SASS variables](https://sass-lang.com/documentation/variables) refer to CSS values set to a name, such as `$font-size`, in SASS files that can not only be referenced after their declaration but can also be used to utilize many of SASS's powerful features. They are compiled at runtime and do not exist in compiled CSS output.
+
+Pros:
+
+- SASS variables have been around for some time and have established best organizational practices.
+- Using SASS variables would allow usage of more advanced SASS functions like calculations, functions like `darken`, and directives like `@if ($some-variable >= 14px)`.
+
+Neutral:
+
+- SASS variables are imperative and can only hold one value at a time so if the variable changes in value, earlier references will remain the same.
+- We'd need to add to our `rescripts` webpack custom config to expose global SASS variables to all of our files.
+
+Cons:
+
+- SASS variables offer a limited set of "pre-compilation" power because variables can't be set dynamically after the stylesheet has been loaded.
+
+#### CSS variables
+
+[CSS custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties), or CSS variables, provide a powerful native browser solution to setting CSS properties to variables. They are declared as `--color: grey` and used as `color: var(--font-color)` and are present in outputted CSS after SASS compilation.
+
+Pros:
+
+- We can still use many SASS features with CSS variables. We will still be able to nest selectors, use mixins that don't require calculations based on variable values such as our responsive design mixins (like `@include min-width('tablet')`) and other mixins with the purpose of shared styles for DRY'ing up our code.
+- Choosing CSS variables helps to "future proof" FxA's front-end stack. Where SASS allows function execution on SASS variables, down the line we will be able to use CSS functions, such as color modifiers, on native CSS variables.
+- CSS variables are easy to target with JavaScript when necessary and unlock new possibilities - while we have chosen to prefer CSS in SCSS files and not in JS (see the second decision made in this ADR), it does open the possibility of updating variable values if the use case arises.
+- CSS variables are ideal for theming and will make a future "dark mode" feature requested by UX very straightforward, as we would simply change a set of variables either in a `prefers-color-scheme` media query or by checking a flag set in localStorage.
+- CSS variables can be global or scoped to an element/selector. For example, variable values can change per class without having to set the property again:
+
+```
+h2 {
+  --h2-color: grey;
+  color: var(--h2-color);
+}
+
+.special h2 {
+  --h2-color: black;
+}
+```
+
+Neutral:
+
+- While CSS variables meet the minimum browser compatibility list for FxA and have [great support otherwise](https://caniuse.com/#feat=css-variables), they aren't supported in IE11.
```

Comment:
We don't officially support IE11 so that's not a real concern.

Reply:
Does this mean we can just nuke the need for a polyfill? I can move this to the pros section if so since they'll just work out of the box without anything else on our end.

Reply:
Heh, I can only tell you what's in https://github.com/mozilla/fxa/blob/master/docs/supported-browsers.md.  Not sure what the implications are for CSS in new Settings.  But yeah, if we are going to do something _only_ for IE11, then we should not.

Reply:
I'm fine with it still being noted here, though -- no harm with it being thought about :)